### ISSUE TITLE
lesson 8 exercise 2 solution does not build

### ISSUE THREAD

**Issue by aptly-io** [NordicDeveloperAcademy/ncs-inter#8]
Using toolchain and SDK v2.6.0 for the nrf5340dk_nrf5340_ns configuration on Ubuntu 22.04.
The build fails and the VSC terminal shows this error (I attached the full log as well):
```
/home/francis/ncs/toolchains/2be090971e/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `bss' will not fit in region `RAM'
/home/francis/ncs/toolchains/2be090971e/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: region `RAM' overflowed by 272256 bytes
```
I would not expect the solution to fail building for the latest SDK

It is unlikely related to above but the terminal also shows this warning:
```
warning: MCUMGR_TRANSPORT_UART (defined at subsys/mgmt/mcumgr/transport/Kconfig.uart:12) was
assigned the value 'y' but got the value 'n'. Check these unsatisfied dependencies: CRC (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MCUMGR_TRANSPORT_UART and/or look up
MCUMGR_TRANSPORT_UART in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.
```
[log.txt](https://github.com/NordicDeveloperAcademy/ncs-inter/files/15045812/log.txt)

**Comment by aptly-io** [comment]
The solution of exercise 3 in lesson 8 fails with a similar error.

**Comment by aptly-io** [comment]
Since exercise 1 succeeds, I added piece by piece parts from the exercise 2 solution to see what changes makes the compilation fail. There's this `chosen` in the app and mcuboot overlay that uses a large space in .bss:
```
/ {
    chosen {
        nordic,pm-ext-flash = &mx25r64;
    };
};
```
Among the many difference between `zephyr_pre0.map` when it succeeds and when it fails building there's this one that jumps out:
```
 .bss.mock_flash
                0x000000002000266a    0x40000 zephyr/drivers/flash/libdrivers__flash.a(flash_simulator.c.obj)
```
Might this be related to the failure?

**Comment by aptly-io** [comment]
By defining a pm_static.yml, as below, in the root of the example it exercise compiles and runs:
```yaml
mcuboot_sram:
  address: 0x20000000
  size: 0x10000
```

I don't understand I run into these issues ... 
1. The MCUboot runs at a moment it has all RAM memory of the nrf5340 for itself. Why then is there a size limit of 0x8000?
2. After compilation, the memory report in VSC shows there's only 19.43K RAM used. That's below this size limit of 32K. Why then did linker step fail telling RAM overflows with 7960?

**Comment by aptly-io** [comment]
Now that everything compiles and links ... Only the serial recovery in MCUboot with the USB works.

While the app is running (blinking LED), the mcumgr-cli times out each time. I fail to list images or upload a new image in an alternative slot while the app is active ...

**Comment by dawidwojtas-mudita** [comment]
@aptly-io Try to add this line in prj.conf
```
# Configure MCUMGR transport to UART
CONFIG_MCUMGR_TRANSPORT_UART=y
CONFIG_CRC=y
```

This line should disappear
>warning: MCUMGR_TRANSPORT_UART (defined at subsys/mgmt/mcumgr/transport/Kconfig.uart:12) was
assigned the value 'y' but got the value 'n'. Check these unsatisfied dependencies: CRC (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MCUMGR_TRANSPORT_UART and/or look up
MCUMGR_TRANSPORT_UART in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.

and everything should work ;)

**Comment by hellesvik-nordic** [comment]
The issue here is that the "_ns" build  adds TF-M.
I think it should be fixed in https://github.com/NordicDeveloperAcademy/ncs-inter/commit/32ec0ca3f3a7517920d41722d7d269cda60ad7de . Let me know if it still does not work

**Comment by aptly-io** [comment]
Thank you for your comments.

@hellesvik-nordic I commented on that fix why I'm not in favor of this unsecure solution.


@dawidwojtas-mudita both UART and USB MCUMgr transport mechanisms are configured. Once USB was added, the UART mechanism stopped working. UART fails both in serial recovery mode (e.g. after pressing the DFU button) and MCUMgr SMP from the application image.
Likely in production this would not be an issue as one would choose to support or UART or USB for updating.
But it would be nice is the course material would explain why it fails and should not suggest both can work together.
